mdstat.go

```diff
@@ -0,0 +1,161 @@
+package procfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	statuslineRE = regexp.MustCompile(`(\d+) blocks .*\[(\d+)/(\d+)\] \[[U_]+\]`)
+	buildlineRE  = regexp.MustCompile(`\((\d+)/\d+\)`)
+)
+
+// MDStatus holds info parsed from /proc/mdstat.
+type MDStatus struct {
+	// Name of the device.
+	MDName string
+	// activity-state of the device.
+	ActivityState string
+	// Number of active disks.
+	DisksActive int64
+	// Total number of disks the device consists of.
+	DisksTotal int64
+	// Number of blocks the device holds.
+	BlocksTotal int64
+	// Number of blocks on the device that are in sync.
+	BlocksSynced int64
+}
+
+// ParseMdstat parses an mdstat-file and returns a struct with the relevant infos.
+func (fs FS) ParseMdstat() (mdstates []MDStatus, err error) {
```

Comment:
I'd still prefer the consistency though over keeping it exactly word-boundaried like the original name. So `ParseMDStat`, but I'd also vote for renaming the `MDStatus` struct to `MDStat`.

Hmm, the other types in this package are all prefixed with `Proc...` at the moment (`ProcStat`, `ProcLimit`, ...), but that's actually an anti-pattern because the name of the package is already `proc`. So maybe don't do that here.